contracts/test/unit/PCVGuardian.sol

```diff
@@ -0,0 +1,491 @@
+
+        getSafeAddresses = pcvGuardian.getSafeAddresses();
+        assertEq(getSafeAddresses.length, 3);
+        assertEq(getSafeAddresses[0], address(pcvDeposit1));
+        assertEq(getSafeAddresses[1], address(pcvDeposit2));
+        assertEq(getSafeAddresses[2], address(pcvDeposit3));
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit1)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit2)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit3)), true);
+
+        vm.prank(addresses.governorAddress);
+        pcvGuardian.unsetSafeAddress(address(pcvDeposit3));
+
+        getSafeAddresses = pcvGuardian.getSafeAddresses();
+        assertEq(getSafeAddresses.length, 2);
+        assertEq(getSafeAddresses[0], address(pcvDeposit1));
+        assertEq(getSafeAddresses[1], address(pcvDeposit2));
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit1)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit2)), true);
+        assertEq(pcvGuardian.isSafeAddress(address(pcvDeposit3)), false);
+    }
+
+    // can't set an already safe address
+    // can't unset an already unsafe address
+    function testStateConsistency2() public {
+        address safeAddress = address(0x42);
+        vm.startPrank(addresses.governorAddress);
+
+        vm.expectRevert("PCVGuardian: not a safe address");
+        pcvGuardian.unsetSafeAddress(safeAddress);
+
+        pcvGuardian.setSafeAddress(safeAddress);
+
+        vm.expectRevert("PCVGuardian: already a safe address");
+        pcvGuardian.setSafeAddress(safeAddress);
+
+        vm.stopPrank();
+    }
+
+    // withdrawals
+
+    // should not be able to withdraw to a non-safe address
+    function testOnlyWithdrawableToSafeAdress() public {
+        vm.prank(addresses.guardianAddress);
+        vm.expectRevert("PCVGuardian: address not whitelisted");
+        pcvGuardian.withdrawToSafeAddress(address(pcvDeposit1), address(pcvDeposit3), 1, false, true);
+
+        vm.prank(addresses.pcvGuardianAdminAddress);
+        pcvGuardian.setSafeAddress(address(pcvDeposit3));
+
+        vm.startPrank(addresses.guardianAddress);
+        pcvGuardian.withdrawToSafeAddress(address(pcvDeposit1), address(pcvDeposit3), 1, false, true);
+        pcvGuardian.withdrawToSafeAddress(address(pcvDeposit3), address(pcvDeposit1), 1, false, true);
+        pcvGuardian.unsetSafeAddress(address(pcvDeposit3));
+        vm.expectRevert("PCVGuardian: address not whitelisted");
+        pcvGuardian.withdrawToSafeAddress(address(pcvDeposit1), address(pcvDeposit3), 1, false, true);
+        vm.stopPrank();
+    }
+
+    // should withdraw from a pcv deposit
+    // should withdrawETH from a pcv deposit
+    // should withdrawERC20 from a pcv deposit
+    // should withdrawRatio from a pcv deposit
+    // should withdrawETHRatio from a pcv deposit
+    // should withdrawERC20Ratio from a pcv deposit
+    function testWithdrawals() public {
```

Comment:
This is far too much for one unit test imo. Each of these withdraw functions should be broken out into their own unit test